src/test/integration/users/User.t.sol

```diff
@@ -237,6 +237,34 @@ contract User is Logger, IDelegationManagerTypes, IAllocationManagerTypes {
         return expectedWithdrawals;
     }
 
+    /// @dev Redelegate to a new operator
+    function redelegate(
+        User newOperator
+    ) public virtual createSnapshot returns (Withdrawal[] memory) {
+        print.method("redelegate", newOperator.NAME_COLORED());
+        Withdrawal[] memory expectedWithdrawals = _getExpectedWithdrawalStructsForStaker(address(this));
```

Comment:
While im on the topic, this is useless because we now return withdrawal roots -- so we can query these after the fact.

Reply:
If that is the purpose, I would say that code should exist at the test level rather in this contract. I think user-type methods should be stupid simple and should be (wherever possible) like just interacting with the contracts directly.

Reply:
Otherwise we start building up business logic in the user contract, and it gets harder to maintain